protocol/configurable-calldata-gas.md

```diff
@@ -0,0 +1,85 @@
+
+Rollups have inherited Ethereum's intrinsic gas formula, but rollups have different constraints for the calldata that they process, particularly the L1's data availability capacity, the rollup's block time, and the rollup's gas limit. These constraints change over time, so the cost of calldata must change accordingly to use L1's data availability capacity without exceeding it. At a minimum, rollups will want to adjust their calldata costs with every L1 hard fork that changes blob capacity, which coincides well with OP Stack forks. But any time a rollup wants to change its gas limit (as `SystemConfig` already allows), it will also want to adjust its calldata costs to account for the new gas limit.
+
+Since Pectra increased L1's data availability capacity to a target of 6 128kb blobs every 12 seconds, it can sustain 64 kb/s of throughput. A single rollup with a two-second block time can only accept 128kb of batches per block. At 40 gas per byte (as of Pectra's [EIP 7623](https://eips.ethereum.org/EIPS/eip-7623)), it would cost 5.12M gas per block for a rollup's calldata to use enough blobs to hit L1's blob target. Most OP Stack chains have gas targets high enough to accept far more than 128kb of calldata per block! This allows blocks to exceed the rollup's constraints without putting any pressure on the base fee to price out this congestion.
+
+| Chain | Gas target | Block time | Underpricing factor | Suggested `calldataGasPerCompressedByte` |
+|-------|------------|------------|---------------------|--------------------------------------|
+| Base | 50M | 2s | ≈ 9.8× | 390 |
+| Unichain | 15M | 1s | ≈ 5.9× | 235 |
+| OP Mainnet | 20M | 2s | ≈ 3.9× | 158 |
+| Ink  | 15M | 2s | ≈ 2.9× | 120 |
+| Mode | 15M | 2s | ≈ 2.9× | 120 |
+
+Because so many OP Stack rollups have underpriced calldata, these chains rely on [batcher sequencer throttling](https://docs.optimism.io/operators/chain-operators/configuration/batcher#batcher-sequencer-throttling) to prevent large backlogs of batches from accumulating. The downside of this approach is that it "breaks the fee market." Users must outbid each other's priority fees in a first-price auction instead of expecting all transactions willing to pay the current base fee to be included in blocks. Many applications do not expect to actively participate in priority fee auctions because EIP 1559 has been so successful at eliminating them, so they experience these periods as a denial of service. (The base fee also plummets to zero during these periods, so even when the throttle is no longer binding, it takes time for the priority fee auction to end.)
+
+Pricing calldata correctly should make batcher sequencer throttling unnecessary and eliminate the priority fee auctions we see today when calldata is congested.
```

Comment:
Sequencer throttling is still necessary because L1 congestion can occur due to spam on other rollups, other disruptions on L1 etc. It's an important backstop solution.